src/embrakes/main.cpp

```diff
@@ -50,65 +49,118 @@ void Main::run()
+            f_brake->sendRetract();
           }
-          Thread::sleep(1000);
-          brake_1->checkHome();
+
+          Thread::sleep(em_brakes_.brake_command_wait_time);
+          m_brake->checkHome();
+          f_brake->checkHome();
+
+          m_brake->checkAccFailure();
+          f_brake->checkAccFailure();
+        } else {
+          if (!m_brake->checkClamped()) {
+            m_brake->sendClamp();
+          }
+          if (!f_brake->checkClamped()) {
+            f_brake->sendClamp();
+          }
+          Thread::sleep(em_brakes_.brake_command_wait_time);
+          m_brake->checkHome();
+          f_brake->checkHome();
+
+          m_brake->checkBrakingFailure();
+          f_brake->checkBrakingFailure();
         }
         break;
       case data::State::kCalibrating:
-        if (brake_1->checkClamped()) {
-          brake_1->sendRetract();
+        if (m_brake->checkClamped()) {
+          m_brake->sendRetract();
+        }
+        if (f_brake->checkClamped()) {
+          f_brake->sendRetract();
         }
-        if (!brake_1->checkClamped()) {
+
+        if (!m_brake->checkClamped() && !f_brake->checkClamped()) {
           em_brakes_.module_status = ModuleStatus::kReady;
           data_.setEmergencyBrakesData(em_brakes_);
         }
-        Thread::sleep(1000);
-        brake_1->checkHome();
+
+        Thread::sleep(em_brakes_.brake_command_wait_time);
+        m_brake->checkHome();
+        f_brake->checkHome();
+        m_brake->checkAccFailure();
+        f_brake->checkAccFailure();
         break;
       case data::State::kAccelerating:
-        brake_1->checkAccFailure();
+        m_brake->checkAccFailure();
+        f_brake->checkAccFailure();
         break;
       case data::State::kNominalBraking:
-        if (!brake_1->checkClamped()) {
-          brake_1->sendClamp();
+        if (!m_brake->checkClamped()) {
+          m_brake->sendClamp();
         }
-        Thread::sleep(1000);
-        brake_1->checkHome();
+        if (!f_brake->checkClamped()) {
+          f_brake->sendClamp();
+        }
+        Thread::sleep(em_brakes_.brake_command_wait_time);
+        m_brake->checkHome();
+        f_brake->checkHome();
+
+        m_brake->checkBrakingFailure();
+        f_brake->checkBrakingFailure();
+        break;
+      case data::State::kEmergencyBraking:
+        if (!m_brake->checkClamped()) {
+          m_brake->sendClamp();
+        }
+        if (!f_brake->checkClamped()) {
+          f_brake->sendClamp();
+        }
+        Thread::sleep(em_brakes_.brake_command_wait_time);
+        m_brake->checkHome();
+        f_brake->checkHome();
 
-        brake_1->checkBrakingFailure();
+        m_brake->checkBrakingFailure();
+        f_brake->checkBrakingFailure();
         break;
       case data::State::kFinished:
-        if (tlm_data_.nominal_braking_command) {
-          if (brake_1->checkClamped()) {
-            brake_1->sendRetract();
+        if (!tlm_data_.nominal_braking_command) {
```

Comment:
It would probably make sense to swap this if/else block around, i.e. check for `tlm_data_.nominal_braking_command` as opposed to `!tlm_data_.nominal_braking_command`.

Reply:
Agree.